examples/common/boilerplate.rs

```diff
@@ -164,7 +164,9 @@ pub fn main_wrapper<E: Example>(
         DeviceUintSize::new(size.width as u32, size.height as u32)
     };
     let notifier = Box::new(Notifier::new(events_loop.create_proxy()));
-    let (mut renderer, sender) = webrender::Renderer::new(gl.clone(), notifier, opts).unwrap();
+    let upload_method = webrender::UploadMethod::PixelBuffer(webrender::VertexUsageHint::Stream);
+    let device = webrender::Device::new(gl.clone(), opts.resource_override_path.clone(), upload_method, None);
```

Comment:
Perhaps instead of ```Renderer::new``` taking a ```Device``` it could be:

```
fn new(device_and_shaders: Option<(Device, Shaders)>, ...)
```

Which would mean that you only need to create the device if you're also pre-creating the shaders. If ```None``` is supplied, we'd create the device internally as it was previously.